src/libs/collect.c

```diff
@@ -1439,12 +1489,14 @@ static void row_activated(GtkTreeView *view, GtkTreePath *path, GtkTreeViewColum
   g_free(text);
 
   if(item == DT_COLLECTION_PROP_TAG || item == DT_COLLECTION_PROP_FOLDERS
+     || item == DT_COLLECTION_PROP_DAY || item == DT_COLLECTION_PROP_TIME
      || item == DT_COLLECTION_PROP_COLORLABEL || item == DT_COLLECTION_PROP_GEOTAGGING
      || item == DT_COLLECTION_PROP_HISTORY ||  item == DT_COLLECTION_PROP_LOCAL_COPY)
     set_properties(d->rule + active); // we just have to set the selection
   else
     update_view(d->rule + active); // we have to update visible items too
 
+  d->skipnextupdate = TRUE;
```

Comment:
What is the reason for this flag? Is `dt_collection_update_query` emitting a signal that in turn causes this module to rebuild its list? If that's the reason, then instead of having this flag and risking to run into a race condition when other parts of dt might cause the signal to be emitted at the same time, just disconnect from the signal, call `dt_collection_update_query` and connect again (using `g_signal_handlers_block_by_func`).

Reply:
When the signal is raised by other parts at the same time, the module should rebuilt its list.
The flag is just there to prevent it once, when something is selected from the list.
If I were to block the signal, there would be no update from another part calling the update concurrently, if I understood correctly.